### DISCUSSION TITLE
feature request: fix rename tab behavior

### DISCUSSION THREAD

**Discussion by bonachea** [furybee/chrome-tab-modifier#283] [Ideas]
_In case of bug report, please fill this template and remove italic lines_
_Otherwise, clear all and type :)_

### Tab Modifier version

**Browser:** `Chrome` 

**Extension version:** `0.22.0`

_Can be found here: chrome://extensions/?id=hcbgadmbdkiilgpifjgcakjehmafcjai_

### Expected behavior

When I right-click a web page and do Rename Tab, I'd expect:
1. The text box that comes up should default to containing the current title (regardless if it came from a rule or the default page title)
1. If I hit cancel, no rule should be created and the page should not reload.
1. If I hit ok, The rule it creates should have a title that matches the name I selected, rather than "Rule created from right-click (URL)" which is completely unreadable in the rule list and makes no sense (after all, I just typed in the label I wanted for this entity)
2. The rule it creates should be appended as the new *highest* priority rule, not *lowest*. This way I can make low priority rules for an entire website (possibly manually in the rule list) and use "rename tab" to rename more specific pages within that site and get the expected behavior. 

### Actual behavior

What I see is an empty text box where hitting either OK or Cancel creates a rule with an unreadable name at lowest priority. 

With the current behavior I have to rearrange rule priority every time I rename tab to get the desired behavior, which should be the default IMO. If you don't agree, please at least add a setting for where the "rename tab" rules are appended as highest priority.

Thanks.

**Comment by MartinLichtblau** [comment]
This could be easily done. Do you still need it?

**Comment by bonachea** [comment]
> This could be easily done. Do you still need it?

yes this would be helpful.

However to be honest the more useful fix would be for issue #186, which is very annoying . I believe that issue has no viable workaround (due to the fact that the fix requested by all of #86 #74 #110 for a means to drop unmatched captures was declined).

**Comment by MartinLichtblau** [comment]
Perhaps yesterdays Pull Request will have solved this. The title won't be changed if the DOM selctor can't be found. But now that I think about it... you want that the title doesn't change if the matching rule is unsuccessful. Thanks for the reply, and let's see how the new version affect this.
